includes/CLI/Plugin_Check_Checks_Command.php

```diff
@@ -0,0 +1,177 @@
+<?php
+/**
+ * Class WordPress\Plugin_Check\CLI\Plugin_Check_Checks_Command
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\CLI;
+
+use WordPress\Plugin_Check\Checker\CLI_Checks_Runner;
+use WordPress\Plugin_Check\Plugin_Context;
+use WordPress\Plugin_Check\Utilities\Plugin_Request_Utility;
+use WP_CLI;
+
+/**
+ * Plugin check checks command.
+ */
+final class Plugin_Check_Checks_Command {
+
+	/**
+	 * Plugin context.
+	 *
+	 * @since n.e.x.t
+	 * @var Plugin_Context
+	 */
+	protected $plugin_context;
+
+	/**
+	 * Output format type.
+	 *
+	 * @since n.e.x.t
+	 * @var string[]
+	 */
+	protected $output_formats = array(
+		'table',
+		'csv',
+		'json',
+	);
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Plugin_Context $plugin_context Plugin context.
+	 */
+	public function __construct( Plugin_Context $plugin_context ) {
+		$this->plugin_context = $plugin_context;
+	}
+
+	/**
+	 * Runs plugin check-checks.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--fields=<fields>]
+	 * : Limit displayed results to a subset of fields provided.
+	 *
+	 * [--format=<format>]
+	 * : Format to display the results. Options are table, csv, and json. The default will be a table.
+	 * ---
+	 * default: table
+	 * options:
+	 *   - table
+	 *   - csv
+	 *   - json
+	 * ---
+	 *
+	 * @subcommand check-checks
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $args       List of the positional arguments.
+	 * @param array $assoc_args List of the associative arguments.
+	 *
+	 * @throws WP_CLI\ExitException Show error if not valid runner.
+	 *
+	 * @SuppressWarnings(PHPMD.NPathComplexity)
+	 * @SuppressWarnings(PHPMD.ExcessiveMethodLength)
+	 */
+	public function check_checks( $args, $assoc_args ) {
+		// Get the Runner.
+		$runner = Plugin_Request_Utility::get_runner();
+
+		// Create the runner if not already initialized.
+		if ( is_null( $runner ) ) {
+			$runner = new CLI_Checks_Runner();
```

Comment:
It's not great that we have to use a runner class here just to get the available checks. I realize there is no other way given the current implementation, but I think that would be relatively straightforward to improve: We should decouple registering the default checks from the runner class.

Since we already have the `Default_Check_Repository` class, which literally carries the term "default" in its name, we may want to incorporate registration of the default checks right into that class? It would make sense to me given the name, and it wouldn't hurt the abstraction, as there's still the `Check_Repository` interface as an agnostic foundation.

I would propose something like this:
* Add a private method `Default_Check_Repository::register_default_checks()`.
* Move all code from `Abstract_Check_Runner::register_checks()`, except the first line, into this new method to register the checks on the same instance.
* Call the new `register_default_checks()` in the constructor of the class.
* Remove `Abstract_Check_Runner::register_checks()`, since it's now only a single line that could just be moved to where the method is called (since it's only used in that one place anyway).

Once we have implemented these changes, the CLI command here could simply use `Default_Check_Repository` and wouldn't need any of the other classes as a workaround.

Reply:
@felixarntz I have implemented as per the comment. But I am confused about the tests for `Default_Check_Repository`. In the tests of this class, it is assumed that no checks are registered by default. But now we have registered checks in the constructor. So all checks are failing. How do you suggest we should proceed?

Reply:
I have updated the tests to reflect new changes.